server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/utils/StarTreeQueryHelper.java

```diff
@@ -146,6 +157,34 @@ private static boolean validateStarTreeMetricSupport(
         return false;
     }
 
+    private static boolean validateKeywordTermsAggregationSupport(
+        CompositeDataCubeFieldType compositeIndexFieldInfo,
+        AggregatorFactory aggregatorFactory
+    ) {
+        if (!(aggregatorFactory instanceof TermsAggregatorFactory termsAggregatorFactory)
+            || aggregatorFactory.getSubFactories().getFactories().length < 1) {
```

Comment:
We can revisit this later post 2.19 - in my benchmarks , i see good improvements even without sub aggs for global terms aggs. For numeric terms, since there is no short-cut optimizations, we should always use star tree.

Reply:
I'm keeping this PR contained with fully tested query shapes as of now. Will keep a TODO to evaluate bucket aggs with sub-aggs post this change.

Reply:
yeah agreed. The ask is only for post 2.19 , we can have data via benchmarks and take the call for terms with no sub aggs. But for numeric terms, we definitely must do it since its guaranteed to be faster as the original collector has no optimizations. 

In my benchmarks it was anywhere between 2x and 5x faster or more for both aggs and successive calls were even faster as they get page cached.